packages/@react-native-windows/cli/postinstall.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ * @ts-check
+ */
+
+const chalk = require('chalk');
+const {rm, cp} = require('shelljs');
+
+const {
+  WritableNpmPackage,
+  findPackage,
+} = require('@react-native-windows/package-utils');
+
+(async () => {
+  await applyRnPatches();
+})();
+
+async function applyRnPatches() {
+  // Replace community CLI version of metro packages with our own, since the
+  // CLI used by RN nightlies pulls in a mismatched version
+  const cliDepsToReplace = [
+    'metro',
+    'metro-config',
+    'metro-core',
+    'metro-react-native-babel-transformer',
+    'metro-resolver',
+    'metro-runtime',
+  ];
+
+  for (const dep of cliDepsToReplace) {
+    await replaceCliDependency(dep);
+  }
+}
+
+async function replaceCliDependency(dependencyName) {
+  const rnPackage = await findPackage('react-native', {
+    searchPath: process.cwd(),
+  });
+  if (!rnPackage) {
+    terminateWithError('Unable to locate "react-native" package');
+  }
```

Comment:
This is not necessarily right, but should likely be right enough for most preview users.

Reply:
Changed to gracefully exit on failure instead of terminate